maestrowf/__init__.py

```diff
@@ -53,3 +60,372 @@ def emit(self, record):
+        if status_data:
+            self._status_data = status_data
+        else:
+            raise ValueError("Status data required to layout a table")
+
+        self._status_table = Table()
+        if study_title:
+            self._status_table.title = "Study: {}".format(study_title)
+
+        # Apply any filters: TODO
+
+        cols = list(self._status_data.keys())
+
+        # Some temporary simple filters to exclude params col in this layout
+        col_filters = ['Params']
+
+        cols = [col for col in cols if col not in col_filters]
+
+        # Setup the column styles
+        for nominal_col_num, col in enumerate(cols):
+
+            if col in list(self._theme_dict.keys()):
+                col_style = col
+            else:
+                if nominal_col_num % 2 == 0:
+                    col_style = 'col_style_1'
+                else:
+                    col_style = 'col_style_2'
+
+            self._status_table.add_column(col,
+                                          style=col_style,
+                                          overflow="fold")
+
+        num_rows = len(self._status_data[cols[0]])
+
+        # Alternate dim rows to differentiate them better
+        for row in range(num_rows):
+            if row % 2 == 0:
+                row_style = 'dim'
+            else:
+                row_style = 'none'
+
+            self._status_table.add_row(
+                *['{}'.format(self._status_data[key][row])
+                  for key in cols],
+                style=row_style
+            )
+
+    def render(self, theme=None):
+        """Do the actual printing"""
+
+        # Apply any theme customization
+        if theme:
+            for key, value in theme.items():
+                self._theme_dict[key] = value
+
+        status_theme = Theme(self._theme_dict)
+
+        _printer = Console(theme=status_theme)
+
+        _printer.print(self._status_table)
+
+    def render_to_str(self, theme=None, width=200):
+        """Capture output to string"""
+
+        # Apply any theme customization
+        if theme:
+            for key, value in theme.items():
+                self._theme_dict[key] = value
+
+        status_theme = Theme(self._theme_dict)
+
+        _printer = Console(theme=status_theme,
+                           file=StringIO(),
+                           width=width)
+        _printer.print(self._status_table)
+
+        return _printer.file.getvalue()
+
+
+class NarrowStatusRenderer(BaseStatusRenderer):
+    def __init__(self, *args, **kwargs):
+        super(NarrowStatusRenderer, self).__init__(*args, **kwargs)
+
+        # Setup default theme
+        self._theme_dict = {
+            "State": "bold red",
+            "Step Name": "bold",
+            "Workspace": "blue",
+            "row_style": "",
+            "row_style_dim": "dim",
+            "background": "grey7"
+        }
+
+    def layout(self, status_data=None, study_title=None, filter_dict=None):
+
+        if status_data:
```

Comment:
In this case, just drop the default arguments -- if data is required, it might be worth checking that it matches a type instead of defaulting it to none.